cmd/soroban-cli/src/commands/contract/init.rs

```diff
@@ -265,18 +265,11 @@ fn copy_contents(from: &Path, to: &Path) -> Result<(), Error> {
             copy_contents(&path, &new_path)?;
         } else {
             if file_exists(&new_path.to_string_lossy()) {
-                //if file is .gitignore, overwrite the file with a new .gitignore file
-                if path.to_string_lossy().contains(".gitignore") {
-                    copy(&path, &new_path).map_err(|e| {
-                        eprintln!(
-                            "Error copying from {:?} to {:?}",
-                            path.to_string_lossy(),
-                            new_path
-                        );
-
-                        e
-                    })?;
-                    continue;
+                if new_path.to_string_lossy().contains(".gitignore") {
+                    append_contents(&path, &new_path)?;
```

Comment:
@chadoh I changed this method a bit from when we were working on it together this morning - I moved the check to see if the path is ".gitigore" or "README.md" here instead of in the `append_contents` method itself so that we don't call that method for every file that's being copied in the loop.

Reply:
Another thing that we may need to handle is that if a user calls `soroban contract init` more than once with the `-f` flag, we'll append the frontend template README contetnst more than once, resulting in duplicate info. This is likely an edge case, but something to consider. i don't have a solution in mind yet 🤔

Reply:
I think it's enough of an edge case that we just don't worry about solving it for now? Now and then people will end up with goofy READMEs when they have goofy workflows ¯\\\_(ツ)_/¯

Reply:
I forgot to update this comment thread - but I ended up solving for this edge case in `append_contents`. I am thinking I'll leave this code in and get this puppy merged! But I totally hear ya that this is an edge for people with goofy workflows, and that's probably not something to worry about.

Reply:
Oh nice! Even better then.